docs/doxygen/collectives.md

```diff
@@ -1,20 +1,150 @@
+A UCC collective operation is a group communication operation among the
+participants of the team. All participants of the team are required to call the
+collective operation. Each participant is represented by the endpoint that is
+unique to the team used for the collective operation. This section provides a
+set of routines for launching, progressing, and completing the collective
+operations.
 
-\b Invocation semantics: The ucc\_collective\_init routine is a non-blocking collective operation to initialize the buffers, operation type, reduction type, and other information required for the collective operation. All participants of the team should call the initialize operation. The routine returns once the participants enter the collective initialize operation. The collective operation is invoked using a ucc\_collective\_post operation. ucc\_collective\_init\_and\_post operation initializes as well as post the collective operation.
+**Invocation semantics**: The ucc\_collective\_init routine is a non-blocking
+collective operation to initialize the buffers, operation type, reduction type,
+and other information required for the collective operation. All participants of
+the team should call the initialize operation. The routine returns once the
+participants enter the collective initialize operation. The collective operation
+is invoked using a ucc\_collective\_post operation.
+ucc\_collective\_init\_and\_post operation initializes as well as post the
+collective operation.
 
-\b Collective type: The collective operation supported by UCC is defined by the enumeration ucc\_coll\_type\_t. It supports three types of collective operations: (a) UCC\_{ALLTOALL,ALLTOALLV, ALLGATHER, ALLGATHERV, ALLREDUCE, REDUCE_SCATTER, REDUCE_SCATTERV, BARRIER} operations where all participants contribute to the results and receive the results (b) UCC\_{REDUCE, GATHER, GATHERV, FANIN} where all participants contribute to the result and one participant receives the result. The participant receiving the result is designated as root. (c) UCC\_{BROADCAST, SCATTER, SCATTERV, FANOUT} where one participant contributes to the result, and all participants receive the result. The participant contributing to the result is designated as root.
+**Collective Type**: The collective operation supported by UCC is defined by the
+enumeration ucc\_coll\_type\_t. The semantics are briefly described here,
+however in most cases it agrees with the semantics of collective operations in
+the popular programming models such as MPI and OpenSHMEM. When they differ, the
+semantics changes are documented. All collective operations operate on the team.
+For the collective operations defined in the enumeration, all participants of
```

Comment:
"participants participate"

Lets rephrase: All team members participate in collective operation.

First part "For the collective operations defined in the enumeration" is unnecessary imho, since we don't support anything else (which is not defined in enumeration)